server/application/application.go

```diff
@@ -206,6 +208,20 @@ func (s *Server) Create(ctx context.Context, q *application.ApplicationCreateReq
 	return updated, nil
 }
 
+// GetNodes returns nodes associated with an application
+func (s* Server) GetNodes(ctx context.Context, q *application.ApplicationQuery) (error) {
+	labelsMap, err := labels.ConvertSelectorToLabelsMap(q.Selector)
```

Comment:
1. I don't think we'll want to reuse the ApplicationQuery as inputs. I think it needs to be something specific to this endpoint

2. Users don't need control over selector when getting nodes of an application.